File: python/sinemessage.py
import math, time, os, sys

width = os.get_terminal_size()[0] - 1
DELAY = 0.1
STEP_INCREASE = 0.2

def main():
    if len(sys.argv) > 1:
        message = sys.argv[1]
    else:
        message = 'Hello!'

    step = 0.0
    while True:  # Main program loop.
        width = os.get_terminal_size()[0] - 1
        multiplier = (width - len(message)) / 2
        sinOfStep = math.sin(step)
        padding = ' ' * int((sinOfStep + 1) * multiplier)
        print(padding + message)
        time.sleep(DELAY)
        step += STEP_INCREASE

try:
    main()
except KeyboardInterrupt:
    print('Sine Message, by Al Sweigart al@inventwithpython.com')

File: typescript/sinemessage.ts
// To get ts types, run: npm i --save-dev @types/readline

namespace SineMessage {
    let width: number = 120
    const DELAY: number = 100;
    const MESSAGE: string = 'Hello, world!';
    const STEP_INCREASE: number = 0.004;

    async function sleep(milliseconds: number): Promise<void> {
        return new Promise(resolve => setTimeout(resolve, milliseconds));
    }

    process.on('SIGINT', function() {
        console.log("\nSine Message by Al Sweigart al@inventwithpython.com");
        process.exit();
    });

    function getTerminalWidth(): number {
        if (process.stdout.isTTY) {
            return process.stdout.columns - 1; // -1 is because Windows puts a newline when printing to the last column
        } else {
            return 80;
        }
    }

    async function main(): Promise<void> {
        let step: number = 0.0;
        while (true) {
            width = getTerminalWidth();
            let multiplier: number = (width - MESSAGE.length) / 2;
            let padding: string = ' '.repeat(Math.floor((Math.sin(step * (180 / Math.PI)) + 1) * multiplier));
            console.log(padding + MESSAGE);
            step += STEP_INCREASE;
            await sleep(DELAY);
        }
    }

    main();
}